src/metadata.ts

```diff
@@ -1,55 +1,94 @@
-import { ZodType, ZodTypeAny } from 'zod';
-import { ZodTypes, isZodType } from './lib/zod-is-type';
+import { ZodType } from 'zod/v4';
+import { ZodTypes, isAnyZodType, isZodType } from './lib/zod-is-type';
 import { ZodOpenAPIMetadata, ZodOpenApiFullMetadata } from './zod-extensions';
 import { isNil, omit, omitBy } from './lib/lodash';
 import { ParameterObject, ReferenceObject, SchemaObject } from './types';
 
 export class Metadata {
+  static collectMetadata(
+    schema: ZodType,
+    metadata?: ZodOpenApiFullMetadata
+  ): ZodOpenApiFullMetadata | undefined {
+    const currentMetadata = schema.meta() as ZodOpenApiFullMetadata | undefined;
+
+    const _internal = {
+      ...currentMetadata?._internal,
+      ...metadata?._internal,
+    };
+
+    const description = currentMetadata?.description ?? schema.description;
+
+    const resultMetadata = {
+      ...currentMetadata,
+      ...(description ? { description } : {}),
+      ...metadata,
+    };
+
+    const totalMetadata = {
+      ...(Object.keys(_internal).length > 0 ? { _internal } : {}),
+      ...(Object.keys(resultMetadata).length > 0 ? resultMetadata : {}),
```

Comment:
Can this be just `...resultMetadata`?

Reply:
Yes it cat. It used to be `resultMetadata.metadata` and I've missed the opportunity